storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -31,40 +32,18 @@ pub struct StateStorePruner {
 }
 
 impl DBPruner for StateStorePruner {
-    /// Find out the first undeleted item in the stale node index.
-    ///
-    /// Seeking from the beginning (version 0) is potentially costly, we do it once upon worker
-    /// thread start, record the progress and seek from that position afterwards.
-    fn initialize(&self) {
-        loop {
-            match self.initialize_least_readable_version() {
-                Ok(least_readable_version) => {
-                    info!(
-                        least_readable_version = least_readable_version,
-                        "[state pruner worker] initialized."
-                    );
-                    self.record_progress(least_readable_version);
-                    return;
-                }
-                Err(e) => {
-                    error!(
-                        error = ?e,
-                        "[state pruner worker] Error on first seek. Retrying in 1 second.",
-                    );
-                    sleep(Duration::from_secs(1));
-                }
-            }
-        }
+    fn name(&self) -> &'static str {
+        STATE_STORE_PRUNER_NAME
     }
 
-    fn prune(&self, max_versions: usize) -> anyhow::Result<Version> {
+    fn prune(&self, max_versions: Version) -> anyhow::Result<Version> {
```

Comment:
why do you prefer `max_versions` to be a `Version` instead of `usize`? It's a number of items and seems proper to be a `usize`

Reply:
@msmouse  - Version is u64, so it makes sense for us to support max_version which is also u64 instead of usize?

Reply:
no I think max_versions is "max number of versions", which is a usize :P

Reply:
Hmm but you might need to prune all the versions in the DB (in extreme case), in that case, you would need max_version to support u64 instead of usize?

Reply:
Is your concern around making this of type `Version` or making it of type `u64` ? I can change this to `u64`, which is semantically same as `Version` but to make it more readable?

Reply:
Made this u64 for clarity.